packages/elements/src/react/sign-in/root.tsx

```diff
@@ -44,7 +44,9 @@ function SignInFlowProvider({ children, exampleMode }: SignInFlowProviderProps)
         actor.send(evt);
       }
     });
-  }, [clerk, exampleMode, formRef, router]);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [clerk, exampleMode, formRef?.id, !!router]);
```

Comment:
I saw that formRef and router were causing many unnecessary re-renders. I narrowed the checks down here and in sign up to optimize.